presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -775,6 +795,64 @@ private void queryCompletedEvent(Optional<ExecutionFailureInfo> failureInfo)
             queryInfoOutputPath.ifPresent(path -> writeQueryInfo(path, queryInfo, queryInfoJsonCodec));
         }
 
+        private void processShuffleStats()
+        {
+            List<PrestoSparkShuffleStats> statsList = new ArrayList<>(shuffleStatsCollector.value());
+            Map<ShuffleStatsKey, List<PrestoSparkShuffleStats>> statsMap = new HashMap<>();
+            Set<Integer> fragments = new HashSet<>();
+            for (PrestoSparkShuffleStats stats : statsList) {
+                ShuffleStatsKey key = new ShuffleStatsKey(stats.getFragmentId(), stats.getOperation());
+                statsMap.computeIfAbsent(key, (ignored) -> new ArrayList<>()).add(stats);
+                fragments.add(stats.getFragmentId());
+            }
+            List<Integer> sortedFragments = new ArrayList<>(fragments);
+            sortedFragments.sort(null);
+            log.info("Shuffle statistics summary:");
+            for (Integer fragment : sortedFragments) {
+                List<PrestoSparkShuffleStats> readStatistics = statsMap.get(new ShuffleStatsKey(fragment, READ));
+                if (readStatistics != null) {
+                    logShuffleStatsSummary(fragment, READ, readStatistics);
+                }
+                List<PrestoSparkShuffleStats> writeStatistics = statsMap.get(new ShuffleStatsKey(fragment, WRITE));
+                if (writeStatistics != null) {
+                    logShuffleStatsSummary(fragment, WRITE, writeStatistics);
+                }
+            }
+        }
+
+        private void logShuffleStatsSummary(int fragmentId, Operation operation, List<PrestoSparkShuffleStats> statsList)
+        {
+            long totalProcessedRows = 0;
+            long totalProcessedBytes = 0;
+            long totalElapsedWallTimeMills = 0;
+            for (PrestoSparkShuffleStats stats : statsList) {
+                totalProcessedRows += stats.getProcessedRows();
+                totalProcessedBytes += stats.getProcessedBytes();
+                totalElapsedWallTimeMills += stats.getElapsedWallTimeMills();
+            }
+            long totalElapsedWallTimeSeconds = totalElapsedWallTimeMills / 1000;
+            long rowsPerSecond = totalProcessedRows;
```

Comment:
Initialize to 0?

Reply:
That's what I had initially. But if the elapsed time is below 1s the output will contain weird `0rows/s` that is confusing. It is better to display `<total_rows>rows/s` in this case that I find less confusing.